api/src/test/java/uk/gov/dhsc/htbhf/claimant/repository/ClaimRepositoryTest.java

```diff
@@ -321,4 +321,25 @@ void shouldThrowExceptionWhenClaimDoesNotExist() {
         assertThat(thrown.getMessage()).contains(claimId.toString());
     }
 
+    @Test
+    void shouldGetClaimsThatHaveBeenInPendingCancellationForMoreThanGivenPeriod() {
+        int numberOfWeeks = 16;
+        LocalDateTime now = LocalDateTime.now();
+        Claim activeClaim = aClaimWithCardStatusAndCardStatusTimestamp(CardStatus.ACTIVE, now.minusWeeks(17));
+        Claim pendingClaimOlderThan16Weeks = aClaimWithCardStatusAndCardStatusTimestamp(CardStatus.PENDING_CANCELLATION, now.minusWeeks(17));
+        Claim pendingClaimLessThan16Weeks = aClaimWithCardStatusAndCardStatusTimestamp(CardStatus.PENDING_CANCELLATION, now.minusWeeks(15));
+        claimRepository.saveAll(List.of(activeClaim, pendingClaimOlderThan16Weeks, pendingClaimLessThan16Weeks));
+
+        List<Claim> claims = claimRepository.getClaimsWithCardStatusPendingCancellationForMoreThan(Period.ofWeeks(numberOfWeeks));
+
+        assertThat(claims).containsOnly(pendingClaimOlderThan16Weeks);
+    }
+
+    @Test
+    void shouldReturnEmptyListWhenThereAreNoClaimsThatHaveBeenInPendingCancellationForMoreThanGivenPeriod() {
```

Comment:
This test would be better if you added some claims to the database, or call it `...NoClaimsInDatabase...`, but I guess that is then a different test so maybe have another test to cover the two scenarios.

Reply:
Added in one claim that's not pending cancellation.